pkg/repository/filter_repository.go

```diff
@@ -1,19 +1,22 @@
 package repository
 
 import (
+	"errors"
+
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcutil/gcs"
 )
 
-type FilterType uint8
+type FilterType byte
 
 const (
 	// only regular filter is supported for now
 	RegularFilter FilterType = iota
 )
 
+var ErrFilterNotFound = errors.New("filter not found")
+
 type FilterRepository interface {
 	PutFilter(*chainhash.Hash, *gcs.Filter, FilterType) error
```

Comment:
I think it's better to change these args with something like this:
```suggestion
        type Filter struct {
            BlockHash []byte
            NBytes []byte
            Type byte
        }
	PutFilter(filter *Filter) error
```

Advantages:
* the struct can be directly stored by any repo implementation. Atm, with badger for example, an ad hoc struct as the one defined would be required to store every info
* If we expected the filter as gcs.Filter, the repo impls may differ in how they serialize it. For example, at first I use repo A, that stores a filter as NBytes. After some time I decide to use Repo B which uses instead PBytes(). Probably the data stored by repo A won't be usable by repo B. We should always store data as raw built-in types to guarantee that the stored data can be used by any impl.

Reply:
Done @altafan, I've finally splitted the struct into `FilterEntry` + `FilterKey` (the idea is to reuse FilterKey in the GetFilter method)